### ISSUE TITLE
Warning building zookeeper with gcc 8.2

### ISSUE THREAD

**Issue by mjwall** [phrocker/sharkbite#17]
With no dependencies installed and letting cmake build them, I got this with gcc version 8.2.0 (Ubuntu 8.2.0-7ubuntu1) 

```
make[4]: Entering directory '/home/user/git/sharkbite/build/dependencies/zookeeper-src'
gcc -DHAVE_CONFIG_H -I.  -I./include -I./tests -I./generated   -Wall -Werror  -g -O2 -D_GNU_SOURCE -MT cli.o -MD -MP -MF .deps/cli.Tpo -c -o cli.o `test -f 'src/cli.c' || echo './'`src/cli.c
mv -f .deps/cli.Tpo .deps/cli.Po
/bin/bash ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I.  -I./include -I./tests -I./generated   -Wall -Werror  -g -O2 -D_GNU_SOURCE -MT zookeeper.lo -MD -MP -MF .deps/zookeeper.Tpo -c -o zookeeper.lo `test -f 'src/zookeeper.c' || echo './'`src/zookeeper.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I./include -I./tests -I./generated -Wall -Werror -g -O2 -D_GNU_SOURCE -MT zookeeper.lo -MD -MP -MF .deps/zookeeper.Tpo -c src/zookeeper.c  -fPIC -DPIC -o .libs/zookeeper.o
src/zookeeper.c: In function ‘format_endpoint_info’:
src/zookeeper.c:3468:21: error: ‘%d’ directive writing between 1 and 5 bytes into a region of size between 0 and 127 [-Werror=format-overflow=]
     sprintf(buf,"%s:%d",addrstr,ntohs(port));
                     ^~
src/zookeeper.c:3468:17: note: directive argument in the range [0, 65535]
     sprintf(buf,"%s:%d",addrstr,ntohs(port));
                 ^~~~~~~
In file included from /usr/include/stdio.h:873,
                 from ./include/zookeeper.h:29,
                 from src/zookeeper.c:27:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:36:10: note: ‘__builtin___sprintf_chk’ output between 3 and 134 bytes into a destination of size 128
   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       __bos (__s), __fmt, __va_arg_pack ());
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[4]: *** [Makefile:990: zookeeper.lo] Error 1
make[4]: Leaving directory '/home/user/git/sharkbite/build/dependencies/zookeeper-src'
make[3]: *** [Makefile:660: all] Error 2
make[3]: Leaving directory '/home/user/git/sharkbite/build/dependencies/zookeeper-src'
make[2]: *** [CMakeFiles/ZooKeeper.dir/build.make:115: ZooKeeper-prefix/src/ZooKeeper-stamp/ZooKeeper-build] Error 2
make[2]: Leaving directory '/home/user/git/sharkbite/build'
make[1]: *** [CMakeFiles/Makefile2:108: CMakeFiles/ZooKeeper.dir/all] Error 2
make[1]: Leaving directory '/home/user/git/sharkbite/build'
make: *** [Makefile:144: all] Error 2
```

Seems like -Wformat-overflow=1 would allow this to continue (https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html).  Any thoughts @phrocker

**Comment by phrocker** [comment]
Sorry I didn't see this until someone E-mailed me. I'm working on a fix now. Should address this and a few other issues across the platform.

There is a bug on zk that hasn't been resolved, but the fix is easy

**Comment by phrocker** [comment]
*this should be resolved.

**Comment by phrocker** [comment]
@chandranitu 

It appears to be the same issue, which I solved here by patching zookeeper in a repo. When I ran into the same issue on hbase-native-client I decided to use the patch command in cmake.

https://github.com/apache/hbase-native-client/blob/master/cmake/DownloadZookeeper.cmake


here is the patch.

https://github.com/apache/hbase-native-client/blob/master/cmake/patches/zookeeper.3.4.x.buf

Since you're working on Apache project, and it supports the cmake ecosystem I would probably aim to solve it by using a patch command. They appear to already have one:

https://github.com/apache/mesos/blob/master/3rdparty/zookeeper-3.4.8.patch

So you might be able to add the patch command to that and see if it helps.
